pkg/resources/management.cattle.io/v3/globalrolebinding/validator.go

```diff
@@ -131,42 +131,38 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		}
 		return nil, fmt.Errorf("unable to get global rules from role %s: %w", globalRole.Name, err)
 	}
-	hasBind, err := a.isRulesAllowed(request, clusterRules, globalRole.Name, a.grbResolver)
-	if err != nil {
-		return admission.ResponseFailedEscalation(err.Error()), nil
-	}
-	if hasBind {
-		// user has the bind verb, no need to check global permissions
-		return admission.ResponseAllowed(), nil
-	}
 
-	rules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(globalRole)
-	_, err = a.isRulesAllowed(request, rules, globalRole.Name, a.resolver)
-	// don't need to check if this request was allowed due to the bind verb since this is the last permission check
-	// if others are added in the future a short-circuit here will be needed like for the clusterRules
+	// Collect all escalations to return to user
+	var returnError error
+	hasBind := common.NewVerbChecker(request, globalRole.Name, bindVerb, a.sar, globalRoleGvr)
+	err = auth.ConfirmNoEscalation(request, clusterRules, "", a.grbResolver)
 	if err != nil {
-		return admission.ResponseFailedEscalation(err.Error()), nil
+		if hasBind("") {
+			return admission.ResponseAllowed(), nil
+		}
+		returnError = errors.Join(returnError, err)
 	}
-	return admission.ResponseAllowed(), nil
-}
-
-// isRulesAllowed checks if the use of requested rules are allowed by the givenResolver for a given request/user
-// returns an error if the user failed an escalation check, and nil if the request was allowed. Also returns a bool
-// indicating if the allow was due to the user having the bind verb
-func (a *admitter) isRulesAllowed(request *admission.Request, rules []rbacv1.PolicyRule, grName string, resolver rbacvalidation.AuthorizationRuleResolver) (bool, error) {
-	err := auth.ConfirmNoEscalation(request, rules, "", resolver)
+	err = auth.ConfirmNoEscalation(request, globalRules, "", a.resolver)
 	if err != nil {
-		hasBind, bindErr := auth.RequestUserHasVerb(request, globalRoleGvr, a.sar, bindVerb, grName, "")
-		if bindErr != nil {
-			logrus.Warnf("Failed to check for the 'bind' verb on GlobalRoles: %v", bindErr)
-			return false, err
+		if hasBind("") {
+			return admission.ResponseAllowed(), nil
 		}
-		if hasBind {
-			return true, nil
+		returnError = errors.Join(returnError, err)
+	}
+	for namespace, rules := range globalRole.NamespacedRules {
+		err = auth.ConfirmNoEscalation(request, rules, namespace, a.resolver)
+		if err != nil {
+			if hasBind(namespace) {
```

Comment:
Similar comment here about checking for bind permissions in a specific namespace - GRs and GRBs are non-namespaced, so we want to check for this at the cluster scope.